src/test/regress/sql/multi_metadata_sync.sql

```diff
@@ -0,0 +1,378 @@
+SELECT master_create_worker_shards('mx_test_table', 8, 1);
+
+-- Set the replication model of the test table to streaming replication so that it is 
+-- considered as an MX table
+UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='mx_test_table'::regclass;
+
+-- Show that the created MX table is included in the metadata snapshot
+SELECT unnest(master_metadata_snapshot());
+
+-- Show that CREATE INDEX commands are included in the metadata snapshot
+CREATE INDEX mx_index ON mx_test_table(col_2);
+SELECT unnest(master_metadata_snapshot());
+
+-- Show that schema changes are included in the metadata snapshot
+CREATE SCHEMA mx_testing_schema;
+ALTER TABLE mx_test_table SET SCHEMA mx_testing_schema;
+SELECT unnest(master_metadata_snapshot());
+
+-- Show that append distributed tables are not included in the metadata snapshot
+CREATE TABLE non_mx_test_table (col_1 int, col_2 text);
+SELECT master_create_distributed_table('non_mx_test_table', 'col_1', 'append');
+UPDATE pg_dist_partition SET repmodel='s' WHERE logicalrelid='non_mx_test_table'::regclass;
+SELECT unnest(master_metadata_snapshot());
+
+-- Show that range distributed tables are not included in the metadata snapshot
+UPDATE pg_dist_partition SET partmethod='r' WHERE logicalrelid='non_mx_test_table'::regclass;
+SELECT unnest(master_metadata_snapshot());
+
+-- Test start_metadata_sync_to_node UDF
+
+-- Ensure that hasmetadata=false for all nodes
+SELECT count(*) FROM pg_dist_node WHERE hasmetadata=true;
+
+-- Run start_metadata_sync_to_node and check that it marked hasmetadata for that worker
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+SELECT nodeid, hasmetadata FROM pg_dist_node WHERE nodename='localhost' AND nodeport=:worker_1_port;
+
+-- Check that the metadata has been copied to the worker
+\c - - - :worker_1_port
+SELECT * FROM pg_dist_local_group;
+SELECT * FROM pg_dist_node ORDER BY nodeid;
+SELECT * FROM pg_dist_partition ORDER BY logicalrelid;
+SELECT * FROM pg_dist_shard ORDER BY shardid;
+SELECT * FROM pg_dist_shard_placement ORDER BY shardid;
+\d mx_testing_schema.mx_test_table
+
+-- Check that pg_dist_colocation is not synced
+SELECT * FROM pg_dist_colocation ORDER BY colocationid;
+
+-- Make sure that truncate trigger has been set for the MX table on worker
+SELECT count(*) FROM pg_trigger WHERE tgrelid='mx_testing_schema.mx_test_table'::regclass;
+
+-- Make sure that start_metadata_sync_to_node considers foreign key constraints
+SET citus.shard_replication_factor TO 1;
+
+CREATE SCHEMA mx_testing_schema_2;
+
+CREATE TABLE mx_testing_schema.fk_test_1 (col1 int, col2 text, col3 int, UNIQUE(col1, col3));
+CREATE TABLE mx_testing_schema_2.fk_test_2 (col1 int, col2 int, col3 text, 
+	FOREIGN KEY (col1, col2) REFERENCES mx_testing_schema.fk_test_1 (col1, col3));
+
+SELECT create_distributed_table('mx_testing_schema.fk_test_1', 'col1');
+SELECT create_distributed_table('mx_testing_schema_2.fk_test_2', 'col1');
```

Comment:
I'm a bit confused about what's going on here, we seem to still be connected to a worker?